### PR TITLE
Fix no index

### LINKED ISSUE #90: dissect crashes on non-indexed samplesheets
Samplesheets without indices (which are valid for single-sample runs, e.g. for MiSeq) cause dissect to crash. Specifically, the error is: 

```
Traceback (most recent call last):                                                                                                                                                             
  File "/localenv/pipegrp/anaconda/miniconda3/envs/dissectBCL/bin/dissect", line 10, in <module>                                                                                               
    sys.exit(dissect())                        
  File "/localenv/pipegrp/anaconda/miniconda3/envs/dissectBCL/lib/python3.10/site-packages/click/core.py", line 1130, in __call__                                                              
    return self.main(*args, **kwargs)                                                          
  File "/localenv/pipegrp/anaconda/miniconda3/envs/dissectBCL/lib/python3.10/site-packages/rich_click/rich_command.py", line 19, in main                                                       
    rv = super().main(*args, standalone_mode=False, **kwargs)                                                                                                                                  
  File "/localenv/pipegrp/anaconda/miniconda3/envs/dissectBCL/lib/python3.10/site-packages/click/core.py", line 1055, in main                                                                  
    rv = self.invoke(ctx)                      
  File "/localenv/pipegrp/anaconda/miniconda3/envs/dissectBCL/lib/python3.10/site-packages/click/core.py", line 1404, in invoke                                                                
    return ctx.invoke(self.callback, **ctx.params)                                             
  File "/localenv/pipegrp/anaconda/miniconda3/envs/dissectBCL/lib/python3.10/site-packages/click/core.py", line 760, in invoke                                                                 
    return __callback(*args, **kwargs)                                                         
  File "/home/pipegrp/src/github.com/maxplanck-ie/dissectBCL/src/dissectBCL/dissect.py", line 38, in dissect                                                                                   
    main(config)                               
  File "/home/pipegrp/src/github.com/maxplanck-ie/dissectBCL/src/dissectBCL/dissect.py", line 97, in main                                                                                      
    exitStats['premux'] = prepConvert(                                                         
  File "/home/pipegrp/src/github.com/maxplanck-ie/dissectBCL/src/dissectBCL/demux.py", line 157, in prepConvert                                                                                
    minP7 = detMask(                           
  File "/home/pipegrp/src/github.com/maxplanck-ie/dissectBCL/src/dissectBCL/demux.py", line 56, in detMask                                                                                     
    minP7 = sampleSheetDF['index'].str.len().min()                                             
  File "/home/pipegrp/.local/lib/python3.10/site-packages/pandas/core/frame.py", line 3458, in __getitem__                                                                                     
    indexer = self.columns.get_loc(key)                                                        
  File "/home/pipegrp/.local/lib/python3.10/site-packages/pandas/core/indexes/base.py", line 3363, in get_loc                                                                                  
    raise KeyError(key) from err                                                               
KeyError: 'index'                              
```

In such cases, the desired behavior is simply to demultiplex using the given sample and continue processing. There will be more samples of this type coming in near future so this is relatively high priority.

### PR DESCRIPTION
**Pull request by dkoppstein**
This PR closes #90. It adds a lot of checks for the index in demux, prepConvert, as well as some general code refactoring. Project 2540 is successfully run using this PR.

### CODE REVIEW

File: src/dissectBCL/demux.py [hunk 1]
```diff
@@ -24,17 +24,18 @@ def misMatcher(P7s, P5s):
...
-    if not P5s.empty and not P5s.isnull().all():
+    for i, ix_list in enumerate((P7s, P5s)):
         hammings = []
-        for comb in combinations(P5s, 2):
-            hammings.append(hamming(comb[0], comb[1]))
-        mmDic['BarcodeMismatchesIndex2'] = hamming2Mismatch(min(hammings))
+        if not ix_list.empty and not ix_list.isnull().all():
+            for comb in combinations(ix_list, 2):
+                hammings.append(hamming(comb[0], comb[1]))
+                mmDic['BarcodeMismatchesIndex{}'.format(i)] = hamming2Mismatch(
```
Comment (WardDeb) [inline src/dissectBCL/demux.py:36]: format(i+1)

Comment (dkoppstein) [reply src/dissectBCL/demux.py]: fixed

File: src/dissectBCL/dissect.py [hunk 2]
```diff
@@ -96,7 +96,8 @@ def sleep():
             )
             exitStats['premux'] = prepConvert(
                 flowcell,
-                sampleSheet
+                sampleSheet,
+                config
```
[the remark this answers is not in the document]
Comment (dkoppstein) [reply src/dissectBCL/dissect.py]: you're right, I got this and postmux mixed up. fixed.

**Comment by WardDeb** [comment]
Common flow cells pass. Will add PR for mask tests after this merger.

**Pull request merged**